### PR TITLE
CC-12700 Enabler

### PR DESCRIPTION
**Pull request by AsonUnique**
- Jira ticket: https://spryker.atlassian.net/browse/CC-12700

### CODE REVIEW

**Review by romanhavrylko — APPROVED** [review]

File: phpstan.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "defaultLevel": 5
```
Comment (abitskil) [inline phpstan.json]: * Raise to 8.

File: LICENSE [hunk 2]
```diff
@@ -0,0 +1,40 @@
+SPRYKER SYSTEMS GMBH EVALUATION LICENSE AGREEMENT
```
Comment (abitskil) [inline LICENSE]: * All `Eco` modules a published under the MIT license.
* Use as a source https://github.com/spryker-eco/oauth-azure/blob/master/LICENSE.

File: composer.json [hunk 3]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "spryker-eco/avalara-tax-extension",
+  "description": "AvalaraTaxExtension module",
+  "license": "proprietary",
```
Comment (abitskil) [inline composer.json]: * MIT.

File: src/SprykerEco/Zed/AvalaraTax/Dependency/Plugin/CreateTransactionRequestExpanderPluginInterface.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+use Generated\Shared\Transfer\CalculableObjectTransfer;
+
+/**
+ * Use this plugin to extend `createTransaction` request before it's is send to the Avalara Tax API.
+ */
+interface CreateTransactionRequestExpanderPluginInterface
+{
+    /**
+     * Specification:
+     * - Expands create transaction request with additional data.
```
Comment (abitskil) [inline src/SprykerEco/Zed/AvalaraTax/Dependency/Plugin/CreateTransactionRequestExpanderPluginInterface.php]: * Expands `createTransaction` request with additional data.

File: phpstan.json (hunk 1 above)
Comment (abitskil) [reply phpstan.json]: Resolved.

File: src/SprykerEco/Shared/AvalaraTax/Transfer/avalara_tax_extension.transfer.xml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0"?>
```
Comment (abitskil) [inline src/SprykerEco/Shared/AvalaraTax/Transfer/avalara_tax_extension.transfer.xml:1]: * Please generate a `pslam` report for this module.

File: LICENSE (hunk 2 above)
Comment (abitskil) [reply LICENSE]: Resolved

File: composer.json (hunk 3 above)
Comment (abitskil) [reply composer.json]: Resolved.

File: composer.json [hunk 6]
```diff
@@ -0,0 +1,30 @@
+{
+  "name": "spryker-eco/avalara-tax-extension",
+  "description": "AvalaraTaxExtension module",
+  "license": "MIT",
+  "require": {
+    "php": ">=7.3",
+    "avalara/avataxclient": "^21.1.1",
```
Comment (abitskil) [inline composer.json]: * Remove the module from dependency.

File: src/SprykerEco/Zed/AvalaraTax/Dependency/Plugin/CreateTransactionRequestExpanderPluginInterface.php [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\AvalaraTax\Dependency\Plugin;
+
+use Generated\Shared\Transfer\AvalaraCreateTransactionRequestTransfer;
+use Generated\Shared\Transfer\CalculableObjectTransfer;
+
+/**
+ * Use this plugin to extend `createTransaction` request before it's is send to the Avalara Tax API.
```
Comment (abitskil) [inline src/SprykerEco/Zed/AvalaraTax/Dependency/Plugin/CreateTransactionRequestExpanderPluginInterface.php]: * Use this plugin to extend `createTransaction` request before it's sent to the Avalara Tax API.

File: src/SprykerEco/Zed/AvalaraTax/Dependency/Plugin/CreateTransactionRequestExpanderPluginInterface.php (hunk 4 above)
Comment (abitskil) [reply src/SprykerEco/Zed/AvalaraTax/Dependency/Plugin/CreateTransactionRequestExpanderPluginInterface.php]: Resolved.

**Pull request merged**